backend/ligand_loader.py

```
@@ -0,0 +1,966 @@
+                        count_before = cursor.fetchone()[0]
+                        
+                        # Load ligands from ChEMBL
+                        result = load_ligand_from_chembl(target_id, chembl_target_id, interaction_type, conn)
+                        if result is not None:
+                            # result is now the count of newly loaded ligands
+                            newly_loaded = result
+                            total_loaded += newly_loaded
+                            if newly_loaded > 0:
+                                print(f"   ✅ Loaded {newly_loaded} new ligands from ChEMBL for target {target_id}")
+                            else:
+                                print(f"   ℹ️  No new ligands loaded for target {target_id} (may already exist)")
```

Ruff (0.15.2)

[warning] String contains ambiguous `ℹ` (INFORMATION SOURCE). Did you mean `i` (LATIN SMALL LETTER I)?

(RUF001)